lib/Sema/CSSimplify.cpp

```diff
@@ -3327,6 +3327,10 @@ ConstraintSystem::simplifyBridgingConstraint(Type type1,
                                              Type type2,
                                              TypeMatchOptions flags,
                                              ConstraintLocatorBuilder locator) {
+  // There's no bridging without ObjC interop.
+  if (!TC.Context.LangOpts.EnableObjCInterop)
```

Comment:
This could be an `assert()`. We should never generate "bridging" constraints unless bridging is enabled.